### PR TITLE
Limit sec scanner reporting to station z

### PR DESCRIPTION
**Pull request by flappybatpal**
[qol]

## About the PR 
Restricts security scanner reporting to the station Z level.

## Why's this needed? 
Security gets pings for things off the station Z such as NT Retention. This doesn't make a lot of sense due to signal range and what are they going to do about it?

### CODE REVIEW

File: code/obj/machinery/secscanner.dm [hunk 1]
```diff
@@ -61,6 +61,8 @@
 			src.use_power(15)
 
 			//////PDA NOTIFY/////
+			if (src.loc.z != Z_LEVEL_STATION)
+				return
```
Comment (ZeWaka) [inline code/obj/machinery/secscanner.dm:65]: you're ignoring the scanner icon being set back, if this was tested you would've noticed 👿

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/machinery/secscanner.dm (hunk 1 above)
Comment (flappybatpal) [reply code/obj/machinery/secscanner.dm:65]: I did this miss but it was tested. Had to find a z level scanner (Owlery, Staff Hall via jump), PDA that showed alerts (sec), make sure my ID didn't have contraband, etc.

I worked out why it's not sticking as red for an item failure, this thing constantly changes icon_state in process(), the change back just makes the timing slightly more consistent.

**Comment by flappybatpal** [comment]
Refactored the PDA notify section into a proc. I believe this behaves as before, have done some tests as werewolf, abomination, mutantraces, player critter and didn't see a change in behaviour.

It does have a change to the power usage. Previously it would use_power once/twice and it seemed pretty arbitrary, an item would use power twice if it was contraband (ok, additional power draw to report maybe), a mob would use power twice if emagged and fails the check but only if human, if not emagged it would use power twice for any mob that failed contraband.

The power usage is a tiny 15 (a door uses 5) so rather than hit the powernet twice for a negligible usage I thought just the single activation cost would be better.

**Review by geneva-breaker — CHANGES_REQUESTED** [review]
Looking this over, this is not the way to do it for futureproofing, and kind of makes the sec scanner more annoying to use for non-station mapping. 

Requesting to scrap the z-level check entirely, instead assign an Network like cameras and such use, and broadcast to PDAs of the appropriate network. This will add value to the scanner, and it's going to be necessary for future Indev stuff anyway, so rather than refactoring that later down the line i'd rather it be done right off the bat.

**Comment by ZeWaka** [comment]
Ditto

**Pull request closed without merging**